components/channel_info_rhs/menu.tsx

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import styled from 'styled-components';
+import {useIntl} from 'react-intl';
+
+import {Channel} from 'mattermost-redux/types/channels';
+import {Constants} from 'utils/constants';
+
+const Icon = styled.div`
+    color: rgba(63, 67, 80, 0.56);
+`;
+
+interface MenuItemProps {
+    className?: string;
+
+    icon: JSX.Element;
+    text: string;
+    onClick: () => void;
+}
+
+const menuItem = ({icon, text, className, onClick}: MenuItemProps) => {
+    return (
+        <div className={className}>
+            <div
+                className='MenuItem--main-action MenuItem--container'
```

Comment:
Since we're using styled components for this component, I'd recommend breaking these subcomponents up into more styled components. The naming kind of looks like BEM conventions which is confusing because it isn't what's intended here.